util/db/cluster.go

```diff
@@ -106,71 +122,79 @@ type ClusterEvent struct {
 	Cluster *appv1.Cluster
 }
 
-// WatchClusters allow watching for cluster events
-func (db *db) WatchClusters(ctx context.Context, callback func(*ClusterEvent)) error {
-	listOpts := metav1.ListOptions{}
-	labelSelector := labels.NewSelector()
-	req, err := labels.NewRequirement(common.LabelKeySecretType, selection.Equals, []string{common.LabelValueSecretTypeCluster})
-	if err != nil {
-		return err
-	}
-	labelSelector = labelSelector.Add(*req)
-	listOpts.LabelSelector = labelSelector.String()
-	w, err := db.kubeclientset.CoreV1().Secrets(db.ns).Watch(listOpts)
-	if err != nil {
-		return err
-	}
-
+func (db *db) WatchClusters(ctx context.Context,
+	namespace string,
```

Comment:
The `db` structure already has namespace (`ns` field). No need to pass it as a parameter